spec/cli_spec.rb

```diff
@@ -120,12 +120,104 @@
         expect(stdout.lines[0]).to eq("F\n")
         expect(stdout).to include <<~ERR
           #{FAIL_APP_DIR}/dumpers/fail_app_dumper.rb:4 fail_app.invalid_status_code received unexpected status code 200 OK (expected 404)
-          #{Dir.getwd}/lib/rails_response_dumper/runner.rb:77:in `block (3 levels) in run_dumps': unexpected status code 200 OK (expected 404) (RuntimeError)
+          #{Dir.getwd}/lib/rails_response_dumper/runner.rb:91:in `block (3 levels) in run_dumps': unexpected status code 200 OK (expected 404) (RuntimeError)
         ERR
         expect(stdout).not_to include(invalid_number_of_statuses)
         expect(stdout).not_to include(dumper_2_invalid_status_code)
         expect(status.exitstatus).to eq(1)
       end
     end
   end
+
+  context 'with filename argument' do
+    context 'without previous dumps' do
+      before { FileUtils.rm_rf("#{APP_DIR}/dumps") }
```

Comment:
Maybe the output should go to a temporary directory so tests are isolated.

Reply:
> Or that each test has its own output directory?

This one.

I would expect this directory would be created with something like `mktmpdir` and cleaned up after the example runs.

https://ruby-doc.org/3.2.1/stdlibs/tmpdir/Dir.html#method-c-mktmpdir

Using a separate directory per tests allows the tests to run in parallel.